Makefile.PL

```diff
@@ -1,9 +1,13 @@
 use ExtUtils::MakeMaker;
 use lib 'inc'; # load our bundled version of Devel::CheckLib
-use Devel::CheckLib;
+
+my $compiler_available;
+BEGIN {
+    $compiler_available = eval { require Devel::CheckLib; 1 };
+}
 
 my %require_mpugmp;
-my $have_gmp = check_lib(lib => 'gmp', header => 'gmp.h');
+my $have_gmp = $compiler_available && check_lib(lib => 'gmp', header => 'gmp.h');
```

Comment:
This doesn't look like it got tested; changing from `use Devel::CheckLib` to `require` loses the import, so you'd need to add the package in calling the function.